tests/testthat/test-migrate.R

```diff
@@ -388,3 +388,305 @@ test_that("migrate() coerces 'character'-type `state` columns to type 'factor'",
+      percent = FALSE,
+      fill_state = iris,
+      verbose = FALSE
+    ),
+    reg_exp = "`fill_state` argument must be length-one value (of type `character`, `numeric`, or `factor`)"
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `character`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "NR",
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `numeric`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = 999,
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `factor`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = as.factor("No Rating"),
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is a value that already exists", {
+
+  migrated_data_without_missing <- migrate(
+    data = mock_credit,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    verbose = FALSE
+  )
+
+  migrated_data_with_missing <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "CCC",
+    verbose = FALSE
+  )
+
+  expect_identical(nrow(migrated_data_without_missing), nrow(migrated_data_with_missing))
+  expect_identical(sum(migrated_data_without_missing$count),sum(migrated_data_with_missing$count))
+
+})
+
+test_that("migrate() outputs more rows when `fill_state` is a value that doesn't exist", {
+
+  migrated_data_without_missing <- migrate(
+    data = mock_credit,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    verbose = FALSE
+  )
+
+  migrated_data_with_missing <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "NR",
+    verbose = FALSE
+  )
+
+  expect_true(nrow(migrated_data_without_missing) < nrow(migrated_data_with_missing))
+
+})
+
+test_that("migrate() sets the value defined in `fill_state` as the smallest factor level", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "ABC",
+    verbose = FALSE
+  )
+
+  levels <- levels(migrated_data$risk_rating_start)
+
+  expect_identical("ABC", levels[length(levels)])
+
+})
```

Comment:
I think we mean to see "... as the *biggest* factor level" ?

The following code

```
devtools::load_all()

mock_credit_with_missing <- mock_credit |> 
  # Remove the first 10 rows
  dplyr::slice(-(1:10)) |>
  # Remove the last 20 rows
  dplyr::slice(-((dplyr::n() - 19):dplyr::n()))

migrated_data <- migrate(
    data = mock_credit_with_missing,
    time = date,
    state = risk_rating,
    id = customer_id,
    percent = FALSE,
    fill_state = "ABC",
    verbose = FALSE
  )

head(migrated_data$risk_rating_start, 0L)
```

returns

```
ordered()
Levels: AAA < AA < A < BBB < BB < B < CCC < ABC
```

where "ABC" is greater than the other factor levels.

Reply:
You are right! Changed applied!